mmdet3d/models/roi_heads/centerpoint_roi_head.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from torch.nn import functional as F
+
+from mmdet3d.core import AssignResult
+from mmdet3d.core.bbox import LiDARInstance3DBoxes, bbox3d2result
+from mmdet.core import build_assigner, build_sampler
+from ..builder import HEADS, build_head, build_roi_extractor
+from .base_3droi_head import Base3DRoIHead
+
+
+@HEADS.register_module()
+class CenterPointRoIHead(Base3DRoIHead):
+    """RoI Head of Two-Stage CenterPoint.
+
+    Args:
+        bev_feature_extractor_cfg (dict): Config dict of BEV feature extractor.
+        bbox_head (dict): Config dict of bbox head.
+        # TODO: refine train_cfg
+        train_cfg (dict): Config of the training.
+        test_cfg (dict, optional): Config of the testing. Defaults to None.
+        init_cfg (dict, optional): Initialization config dict. Defaults to None
+    """
+
+    def __init__(self,
+                 bev_feature_extractor_cfg,
+                 bbox_head,
+                 train_cfg=None,
+                 test_cfg=None,
+                 pretrained=None,
+                 init_cfg=None):
+        super(CenterPointRoIHead, self).__init__(
+            bbox_head=bbox_head,
+            train_cfg=train_cfg,
+            test_cfg=test_cfg,
+            pretrained=pretrained,
+            init_cfg=init_cfg)
+        # build a bev feature extractor
+        self.bev_feature_extractor = build_roi_extractor(
+            bev_feature_extractor_cfg)
+
+    def init_bbox_head(self, bbox_head):
+        """Initialize box head.
+
+        Args:
+            bbox_head (dict): Config dict of RoI Head.
+        """
+
+        self.bbox_head = build_head(bbox_head)
+
+    def init_mask_head(self):
+        """Initialize maek head."""
+        pass
+
+    def init_assigner_sampler(self):
+        """Initialize assigner and sampler."""
+        self.bbox_assigner = None
+        self.bbox_sampler = None
+        if self.train_cfg:
```

Comment:
if :
# keep the original
else:
        self.bbox_assigner = None
        self.bbox_sampler = None